### PR TITLE
migrate setup.py to setup.cfg

### PR DESCRIPTION
**Pull request by martinobersteiner**
migrate from `setup.py` to `setup.cfg`
add `pyproject.toml`
~~use `pylint` from now on~~ (-> moved to different PR)
~~fix various things as per suggestion of pylint~~

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[build-system]
+requires = ["setuptools", "wheel"]
+build-backend = "setuptools.build_meta"
```
Comment (utnapischtim) [inline pyproject.toml:3]: please add a newline to the last line of the file. (this should be configurable in the editor/IDE)

File: invenio_records_lom/cli.py [hunk 2]
```diff
@@ -10,15 +10,51 @@
...
 @click.group()
+@click.version_option("0.1", prog_name="invenio-records-lom cli")
 def lom():
     """CLI-group for "invenio lom" commands."""
-    pass
+
+
+@lom.command("add-indices")
+@with_appcontext
+def add_indices():
```
Comment (utnapischtim) [inline invenio_records_lom/cli.py:28]: This should not be in this PR. Create a separate PR for that CLI command.

File: invenio_records_lom/config.py [hunk 3]
```diff
@@ -26,6 +26,8 @@
     "json": {
         "name": _("JSON"),
         "serializer": "flask_resources.serializers:JSONSerializer",
+        "content-type": "application/json",
```
Comment (utnapischtim) [inline invenio_records_lom/config.py:29]: To which task this change belong? (the same applies to the line below)

File: invenio_records_lom/resources/serializers/__init__.py [hunk 4]
```diff
@@ -10,7 +10,7 @@
 
 from flask_resources.serializers import MarshmallowJSONSerializer
 from invenio_rdm_records.resources.serializers import UIJSONSerializer
-from lxml.builder import ElementMaker
+from lxml.builder import ElementMaker  # pylint: disable=no-name-in-module
```
Comment (utnapischtim) [inline invenio_records_lom/resources/serializers/__init__.py:13]: Is this really necessary? I thought that such disable statements are only necessary if pylint runs outside of a virtualenv where the packages are not installed?

File: invenio_records_lom/resources/serializers/schemas.py [hunk 5]
```diff
@@ -135,7 +135,9 @@ class LOMUIObjectSchema(Schema):
 
     location = Location(attribute="metadata.technical.location")
 
-    rights = Rights(attribute="metadata.rights")
+    rights = Rights(attribute="metadata.rights.description.langstring")
```
Comment (utnapischtim) [inline invenio_records_lom/resources/serializers/schemas.py:138]: What is the reason for this change?

File: invenio_records_lom/ui/records/records.py [hunk 6]
```diff
@@ -128,24 +128,22 @@ def record_export(
...
-        is_preview=is_preview,
-        is_draft=record._record.is_draft,
-    )
+    content_type = exporter.get("content-type", export_format)
+    filename = exporter.get("filename", export_format).format(id=pid_value)
+    headers = {
+        "Content-Type": content_type,
+        "Content-Disposition": f"attachment; filename={filename}",
+    }
+    return (exported_record, 200, headers)
```
Comment (utnapischtim) [inline invenio_records_lom/ui/records/records.py:137]: Does this change belong to the migration from setup.py to setup.cfg? It seams that it belongs to the export feature?

File: pyproject.toml (hunk 1 above)
Comment (martinobersteiner) [reply pyproject.toml]: Added. (I should've figured vscode has a setting for that - thanks)

On a side note, both `invenio-imoox` and `invenio-records-marc21` are missing this newline too...

File: invenio_records_lom/cli.py (hunk 2 above)
Comment (martinobersteiner) [reply invenio_records_lom/cli.py]: Removed.

File: invenio_records_lom/config.py (hunk 3 above)
Comment (martinobersteiner) [reply invenio_records_lom/config.py]: This is part of the export-feature update (see further down comment).
Removed.

File: invenio_records_lom/resources/serializers/__init__.py (hunk 4 above)
Comment (martinobersteiner) [reply invenio_records_lom/resources/serializers/__init__.py:13]: For performance reasons, some of `lxml`'s code is written in C.
C-code added attributes don't get picked up by pylint (not even if pylint is inside a fitting environment).

File: invenio_records_lom/resources/serializers/schemas.py (hunk 5 above)
Comment (martinobersteiner) [reply invenio_records_lom/resources/serializers/schemas.py:138]: Looking at the code of `Rights` (which this PR does not change), `Rights` clearly needs a langstring as input (which `metadata.rights` isn't).
As I see it, the unchanged code couldn't really have worked (it sure didn't for me).

File: invenio_records_lom/ui/records/records.py (hunk 6 above)
Comment (martinobersteiner) [reply invenio_records_lom/ui/records/records.py]: Belongs to the export-feature update.
Removed.

File: invenio_records_lom/resources/serializers/schemas.py (hunk 5 above)
Comment (martinobersteiner) [reply invenio_records_lom/resources/serializers/schemas.py:138]: The following shows the strucutre of `metadata.rights` (due to LOM-UIBK):
image
Note that the CC-url is situated at `metadata.rights.description.langstring.#text`.

However, before this change `Rights` was passed `metadata.rights` which it then took the `.url`-attribute of - so `metadata.rights.url` in total.
As I see it, the unchanged code couldn't really have worked (at least it failed when I tested it).

**Review by utnapischtim — APPROVED** [review]
